testtools/matchers/_impl.py

```diff
@@ -53,7 +62,49 @@ def __str__(self):
         raise NotImplementedError(self.__str__)
 
 
-class Mismatch(object):
+Matcher = IMatcher
+
+
+class IMismatch(object):
+    """A mismatch detected by an IMatcher.
+
+    Describes the protocol that mismatches are required to implement.
+    """
+
+    # XXX: jml would like to extend this interface to include the thing that
```

Comment:
Ah, interesting. I mostly want it for composition, e.g. to help with `Not`. ISTR wanting it for https://github.com/ClusterHQ/flocker/blob/master/flocker/testtools/matchers.py but looking at it, I can't see what was essential.

Whether we or not we extend the interface this way, I want second opinions on whether it should be done in this patch / this release.